crowbar_framework/lib/crowbar/deployment_queue.rb

```diff
@@ -19,9 +19,9 @@ class DeploymentQueue
 
     attr_reader :logger, :proposal_queue
 
-    def initialize(logger: Rails.logger, queue: ProposalQueue.new)
+    def initialize(logger: Rails.logger)
       @logger         = logger
-      @proposal_queue = queue
+      @proposal_queue = ::ProposalQueue.first
```

Comment:
But that is never guaranteed, so please add it directly here or create a named scope ```ordered``` that you can call for get the FIFO record for sure, e.g.

```
  scope :ordered, lambda {
    order("proposal_queue.created_at ASC")
  }
```